service/src/test/java/bio/terra/cbas/util/methods/TestBackfillOriginalWorkspaceIds.java

```diff
@@ -0,0 +1,280 @@
+import bio.terra.cbas.dao.RunSetDao;
+import bio.terra.cbas.models.CbasRunSetStatus;
+import bio.terra.cbas.models.Method;
+import bio.terra.cbas.models.MethodVersion;
+import bio.terra.cbas.models.RunSet;
+import bio.terra.cbas.util.BackfillOriginalWorkspaceIds;
+import java.time.OffsetDateTime;
+import java.util.Arrays;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+@SpringBootTest
+public class TestBackfillOriginalWorkspaceIds {
+  @MockBean private RunSetDao runSetDao;
+  @MockBean private MethodDao methodDao;
+  @MockBean private MethodVersionDao methodVersionDao;
+  @MockBean private CbasContextConfiguration cbasContextConfig;
+  private Logger logger = LoggerFactory.getLogger(TestBackfillOriginalWorkspaceIds.class);
+
+  @BeforeEach
+  void setupContext() {
+    when(cbasContextConfig.getWorkspaceId()).thenReturn(currentWorkspaceId);
+    when(cbasContextConfig.getWorkspaceCreatedDate()).thenReturn(workspaceCreatedDate);
+    when(methodDao.getMethods())
+        .thenReturn(
+            Arrays.asList(
+                methodClonedNull,
+                methodCreatedNull,
+                methodClonedBackfilled,
+                methodCreatedBackfilled));
+    when(methodDao.updateOriginalWorkspaceId(any(), any())).thenReturn(1);
+    when(runSetDao.getRunSets(eq(null), anyBoolean()))
+        .thenReturn(
+            Arrays.asList(
+                runSetClonedNull,
+                runSetCreatedNull,
+                runSetClonedBackfilled,
+                runSetCreatedBackfilled));
+    when(runSetDao.updateOriginalWorkspaceId(any(), any())).thenReturn(1);
+    when(methodVersionDao.getMethodVersions())
+        .thenReturn(
+            Arrays.asList(
+                methodVersionClonedNull,
+                methodVersionCreatedNull,
+                methodVersionClonedBackfilled,
+                methodVersionCreatedBackfilled));
+    when(methodVersionDao.updateOriginalWorkspaceId(any(), any())).thenReturn(1);
+  }
+
+  @Test
+  void testBackfillMethods() {
+    BackfillOriginalWorkspaceIds.backfillMethods(methodDao, cbasContextConfig, logger);
+
+    verify(methodDao).updateOriginalWorkspaceId(methodClonedNull.methodId(), nullWorkspaceId);
+    verify(methodDao).updateOriginalWorkspaceId(methodCreatedNull.methodId(), currentWorkspaceId);
+    verify(methodDao, never())
+        .updateOriginalWorkspaceId(eq(methodClonedBackfilled.methodId()), any());
+    verify(methodDao, never())
+        .updateOriginalWorkspaceId(eq(methodCreatedBackfilled.methodId()), any());
+  }
+
+  @Test
+  void testBackfillRunSets() {
+    BackfillOriginalWorkspaceIds.backfillRunSets(runSetDao, cbasContextConfig, logger);
+    // verify(runSetDao, times(2)).getRunSets(null, false);
```

Comment:
This `verify` call is showing inconsistent behavior for reasons that I don't understand.

When this test is run by itself (in IntelliJ), runSetDao.getRunSets is called twice. My working theory is that `when(runSetDao.getRunSets)...` counts as one call.

When this test runs as part of the test class, runSetDao.getRunSets is called only once! I have no idea why.